Issues 333 334 - fix 1.0.0-rc.1 extensions and 1.0.0-beta.1 schema links

jonhealy1 opened the pull request:
Resolves: https://github.com/stac-utils/pystac/issues/333
Resolves: https://github.com/stac-utils/pystac/issues/334

**Related Issue(s):** #333 #334 

**Description:**
This fixes how version 1.0.0-rc.1 extensions are handled because they are still using the shortened identifiers. It also fixes the schemas for 1.0.0-beta.1 because these are not on schemas.stacspec.org.

**PR Checklist:**

- [x] Code is formatted (run `scripts/format`)
- [ ] Tests pass (run `scripts/test`) - I need a password for this?
- [x] This PR maintains or improves overall codebase code coverage.
- [ ] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac/blob/develop/CHANGELOG.md). See [the docs](https://pystac.readthedocs.io/en/latest/contributing.html#changelog) for information about adding to the changelog.

lossyrob commented:
Hey @jonhealy1, seems like there's a lot of commits here but the final line changes are +12/-8 ... did some things get lost in the shuffle or are some of these changes reverted?

jonhealy1 commented:
Hi @lossyrob sorry yes. I was implementing tests to run validation against some of the stac objects we test against for stac-validator. I pulled those changes out and pushed them to a local branch because they are not finished and I wasn't sure if it was something that would be valuable.

jonhealy1 commented:
There are only a few minor changes in this pr.

lossyrob commented:
Ok, gotcha. I'll do a squash commit on this PR so that we can get rid of those extraneous commits in the main history. GitHub Actions seems to be having some issues so we'll wait until we can rerun the CI, but this is looking good to me

jonhealy1 commented:
Thank you. Next time I will be more careful.

jonhealy1 commented:
@lossyrob I could redo the pr too if it's easier.

lossyrob commented:
Nah a squash commit will do the trick. If you wanted to you could rebase these commits into one or two and force push to your branch, which would clean this PR up, but I wouldn't worry about it too much!

lossyrob commented:
@jonhealy1 I think it would make sense to add a test case for an .rc1 STAC item that would fail without this change, so we can guard against changes that might break this in the future.

jonhealy1 commented:
v1.0.0-rc.2 - 2021-03-30
Changed
Required properties of type string require a minimum length of 1. (#1065)
gsd must be greater than 0. (#1068)
Removed
The remaining extensions in the spec (EO, Projection, Scientific Citation and View) have been moved out of the core specification, into their own repos in the stac-extensions GitHub organization. They must now be referred to by their schemas directly - the name shortcuts for them are no longer valid. The concept of the 'named shortcut' also goes away entirely. (#1070)
Fixed
Examples
Collection Assets were specified as required (only in written text, not in JSON Schema), but that was incorrectly copied over from the former collection-assets extension. Collection Assets are not required.
Clarified that the values in summaries (both for ranges and sets of values) must follow the data type of the property they summarize. (#1069)

jonhealy1 commented:
@lossyrob test_identify.py fails because of the 1.0.0-rc.1 example I added to example-info.csv for testing? 

E               AssertionError: False is not true : Failed /Users/comp/Sparkgeo/pystac/tests/data-files/examples/1.0.0-RC1/extended-item.json:

tests/serialization/test_identify.py:45: AssertionError

jonhealy1 commented:
nevermind - someone put the 1.0.0-beta.2 examples in the 1.0.0-RC1 folder

jonhealy1 commented:
I think serialization/migrate.py is broken and not working

FAILED tests/serialization/test_identify.py::IdentifyTest::test_identify - AssertionError: Items in t...
ERROR tests/test_item.py::test_to_from_dict
ERROR tests/extensions/test_eo.py::test_to_from_dict
ERROR tests/extensions/test_file.py::test_to_from_dict
ERROR tests/extensions/test_label.py::test_to_from_dict
ERROR tests/extensions/test_pointcloud.py::test_to_from_dict
ERROR tests/extensions/test_projection.py::test_to_from_dict
ERROR tests/extensions/test_timestamps.py::test_to_from_dict
ERROR tests/extensions/test_view.py::test_to_from_dict

duckontheweb commented on tests/data-files/examples/example-info.csv:
```diff
@@ -123,6 +123,16 @@
 "hand-0.8.1/collection.json","COLLECTION","0.8.1",""
+"1.0.0-RC1/extended-item.json","ITEM","1.0.0-rc.1","eo|view|projection|scientific"
+"1.0.0-RC2/extended-item.json","ITEM","1.0.0-rc.2","https://stac-extensions.github.io/remote-data/v1.0.0/schema.json|https://stac-extensions.github.io/view/v1.0.0/schema.json|https://stac-extensions.github.io/projection/v1.0.0/schema.json|https://stac-extensions.github.io/scientific/v1.0.0/schema.json|https://stac-extensions.github.io/eo/v1.0.0/schema.json"
+"1.0.0-RC3/catalog.json","CATALOG","1.0.0-rc.3",""
+"1.0.0-RC3/collection-only/collection.json","COLLECTION","1.0.0-rc.3","https://stac-extensions.github.io/eo/v1.0.0/schema.json|https://stac-extensions.github.io/projection/v1.0.0/schema.json|https://stac-extensions.github.io/view/v1.0.0/schema.json"
+"1.0.0-RC3/collection.json","COLLECTION","1.0.0-rc.3","https://stac-extensions.github.io/eo/v1.0.0/schema.json|https://stac-extensions.github.io/view/v1.0.0/schema.json"
+"1.0.0-RC3/collectionless-item.json","ITEM","1.0.0-rc.3","https://stac-extensions.github.io/eo/v1.0.0/schema.json|https://stac-extensions.github.io/view/v1.0.0/schema.json"
+"1.0.0-RC3/core-item.json","ITEM","1.0.0-rc.3",""
+"1.0.0-RC3/extended-item.json","ITEM","1.0.0-rc.3","https://stac-extensions.github.io/eo/v1.0.0/schema.json|https://stac-extensions.github.io/projection/v1.0.0/schema.json|https://stac-extensions.github.io/scientific/v1.0.0/schema.json|https://stac-extensions.github.io/view/v1.0.0/schema.json|https://stac-extensions.github.io/remote-data/v1.0.0/schema.json"
+"1.0.0-RC3/extensions-collection/collection.json","COLLECTION","1.0.0-rc.3",""
+"1.0.0-RC3/extensions-collection/proj-example/proj-example.json","ITEM","1.0.0-rc.3","https://stac-extensions.github.io/eo/v1.0.0/schema.json|https://stac-extensions.github.io/projection/v1.0.0/schema.json"
```
Looks like these files don't exist yet, but once that's fixed this looks good to me, too.

duckontheweb replied:
Actually, I think these just need to be changed to `1.0.0-RC2`. @jonhealy1 I'm happy to merge in `main` and update this if you want. I think this PR is ready to go other than this.

jonhealy1 replied:
I think I only added in one rc1 and one rc2 entry. I think the rc3 stuff comes from a previous iteration of what I was building off of? If you can get it working to merge that would be great. Cheers.

duckontheweb replied:
Ah, yeah, I see that now. I'll try to work through that and get the tests working again.

duckontheweb commented:
@jonhealy1 @lossyrob I didn't have write permissions to this branch, so I opened up a separate PR (#455).
